src/components/Views/CameraView/CameraView.js

```diff
@@ -101,87 +129,239 @@ class CameraView extends React.Component<
         const data = await this.camera.takePictureAsync(options);
         this.setState({ loading: false });
 
-        if (selectedObservation) {
-          updateObservation({
-            ...selectedObservation,
-            media: selectedObservation.media.concat([
-              {
-                type: 'Photo',
-                source: data.uri
-              }
-            ])
+        if (fromEditor) {
+          if (selectedObservation) {
+            updateObservation({
+              ...selectedObservation,
+              media: selectedObservation.media.concat([
+                {
+                  type: 'Photo',
+                  source: data.uri
+                }
+              ])
+            });
+            goToObservationEditor();
+          }
+        } else {
+          goToCategories();
+          const initialObservation = applyObservationDefaults({
+            id: size(observations) + 1,
+            createdFrom: 'camera'
           });
-          goToObservationEditor();
+          createObservation(initialObservation);
+
+          navigator.geolocation.getCurrentPosition(
```

Comment:
We shouldn't have to get current position. This is set in the initial create flow through ducks.